packit_service/worker/reporting.py

```diff
@@ -240,7 +254,56 @@ def report_status_by_comment(
     def get_statuses(self):
         self.project_with_commit.get_commit_statuses(commit=self.commit_sha)
 
-    def comment(self, body: str):
+    def _has_identical_comment(self, body: str, mode: DuplicateCheckMode) -> bool:
+        """Checks if the body is the same as the last or any comment based on mode."""
+        if self.pr_id:
```

Comment:
Does anyone possibly see a better way of doing this? Since the interface of PR and commit comment is different, all solutions became more complicated than this slight duplication of logic, so I ended up doing it this way.